### PR TITLE
GUI: Chosen user/role should be fully into line with entered group or role after press 'enter' key

### PR DESCRIPTION
**Pull request by ZMaratovna**
This PR implements fix for #2123

### CODE REVIEW

File: client/src/components/runs/logs/forms/ShareWithForm.js [hunk 1]
```diff
@@ -137,10 +96,28 @@ export default class ShareWithForm extends React.Component {
...
+     return -1;
+   }
+   if (
+     str2.toLowerCase().indexOf(query) === str1.toLowerCase().indexOf(query) &&
+     str1.toLowerCase().indexOf(query) >= 0
+   ) {
+     return str1.toLowerCase() > str2.toLowerCase() ? 1 : -1;
+   }
+   return 0;
+ }
```
Comment (rodichenko) [inline client/src/components/runs/logs/forms/ShareWithForm.js]: No need to have this function as a method of `ShareWithForm` instance - put it outside of the class

File: client/src/components/runs/logs/forms/ShareWithForm.js [hunk 2]
```diff
@@ -169,30 +147,25 @@ export default class ShareWithForm extends React.Component {
...
   };
 
   closeFindUserDialog = () => {
-    this.setState({findUserVisible: false});
+    this.setState({findUserVisible: false, userSearchString: null});
   };
 
   onSelectUser = async () => {
-    await this.grantPermission(this.selectedUser, true);
+    await this.grantPermission(this.state.userSearchString.trim(), true);
```
Comment (rodichenko) [inline client/src/components/runs/logs/forms/ShareWithForm.js]: We should check if `this.state.userSearchString` is `null`/`undefined`

File: client/src/components/runs/logs/forms/ShareWithForm.js [hunk 3]
```diff
@@ -201,25 +174,30 @@ export default class ShareWithForm extends React.Component {
   };
 
   onSelectGroup = async () => {
+    const {groupSearchString} = this.state;
     const [role] = this.props.roles
-      .filter(r => !r.predefined && this.splitRoleName(r.name) === this.selectedGroup);
-    const roleName = role ? role.name : this.selectedGroup;
+      .filter(r => !r.predefined && this.splitRoleName(r.name) === groupSearchString.trim());
+    const roleName = role ? role.name : groupSearchString.trim();
```
Comment (rodichenko) [inline client/src/components/runs/logs/forms/ShareWithForm.js]: We should check if `groupSearchString` is `null`/`undefined`

**Review by rodichenko — CHANGES_REQUESTED** [review]

File: client/src/components/runs/logs/forms/ShareWithForm.js (hunk 3 above)
Comment (ZMaratovna) [reply client/src/components/runs/logs/forms/ShareWithForm.js]: Done!

File: client/src/components/runs/logs/forms/ShareWithForm.js (hunk 2 above)
Comment (ZMaratovna) [reply client/src/components/runs/logs/forms/ShareWithForm.js]: Done!

File: client/src/components/runs/logs/forms/ShareWithForm.js (hunk 1 above)
Comment (ZMaratovna) [reply client/src/components/runs/logs/forms/ShareWithForm.js]: Done!

**Review by rodichenko — APPROVED** [review]

**Pull request closed without merging**